src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -80,6 +94,10 @@
                 (! result.containsKey("accept-encoding"))) {
             result.put("accept-encoding", new BasicHeader("Accept-Encoding", "gzip, deflate"));
         }
+        if (options.getCompressRequestBody() == CompressType.GZIP &&
+                (! result.containsKey("content-encoding"))) {
+            result.put("content-encoding", new BasicHeader("Content-Encoding", "gzip"));
+        }
```

Comment:
Ah ok, if that's how this is used then it seems good to defer to the value that's already there